botocore/session.py

```diff
@@ -962,6 +963,12 @@ def create_client(
             smart_defaults_factory.merge_smart_defaults(
                 config_store, defaults_mode, region_name
             )
+
+        self._add_configured_endpoint_provider(
+            client_name=service_name,
+            config_store=config_store,
+        )
+
```

Comment:
This doesn't seem right to me. At the very least you should be able to manually add a new section to the config mapping in `create_botocore_default_config_mapping`.  Is there any reason in particular you didn't go that route? We should avoid mutating the config store after initialization and instead make sure to include this new provider along with all the others in `_register_config_store`.

Reply:
This is modeled after the smart defaults behavior in order to have access to all of the data needed. If we did this in the default mapping, we would need to know every service that could possibly exist and add logic for it; this is not known at that time (need a session for that, which introduces a circular dependency).